File: src/models/conversation.py
# src/models/conversation.py

from datetime import datetime
import pytz

class Message:
    def __init__(self, role: str, content: str):
        self.role = role
        self.content = content
        self.timestamp = datetime.now(pytz.UTC)

    def to_dict(self):
        return {
            "role": self.role,
            "content": self.content,
            "timestamp": self.timestamp.isoformat()
        }

class Conversation:
    def __init__(self, id: str, title: str = None):
        self.id = id
        self.title = title or "New Conversation"
        self.messages = []
        self.created_at = datetime.now(pytz.UTC)

    def add_message(self, role: str, content: str):
        message = Message(role, content)
        self.messages.append(message)
        return message

    def get_messages(self):
        return [msg.to_dict() for msg in self.messages]

File: commandr/backend/models/conversation.go
package models

import (
    "time"
    "github.com/google/uuid"
)

type Message struct {
    ID        string    `json:"id"`
    Role      string    `json:"role"`
    Content   string    `json:"content"`
    Timestamp time.Time `json:"timestamp"`
}

type Conversation struct {
    ID        string    `json:"id"`
    Title     string    `json:"title"`
    Messages  []Message `json:"messages"`
    CreatedAt time.Time `json:"createdAt"`
}

func NewConversation(title string) *Conversation {
    return &Conversation{
        ID:        uuid.New().String(),
        Title:     title,
        Messages:  make([]Message, 0),
        CreatedAt: time.Now(),
    }
}

func (c *Conversation) AddMessage(role, content string) Message {
    msg := Message{
        ID:        uuid.New().String(),
        Role:      role,
        Content:   content,
        Timestamp: time.Now(),
    }
    c.Messages = append(c.Messages, msg)
    return msg
}
